<a href="https://colab.research.google.com/github/MDrakib-uddin/flyrank-ml-internship/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# 1. My Lane

I selected **Lane 2: Refresh / Content Opportunity Scoring**.

I chose this lane because the goal is to help content teams decide which pages should be reviewed first when time and resources are limited. Rather than trying to replace human judgment, I want to build a decision-support system that ranks pages based on observable search and engagement signals. This lane aligns well with the starter dataset and provides a practical foundation for the remaining weeks of the internship.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# 2. The Question

### Research Question

Which content pages should be prioritized for review based on observable search and engagement signals?

### Decision

The project helps decide which pages should be reviewed first for content refresh.

### Who takes the action?

SEO specialists or content editors review the highest-ranked pages and decide whether to update titles, improve content quality, expand information, or improve internal linking.

### Cost of a Wrong Recommendation

**False Positive:** Time is spent reviewing pages that already perform well, reducing team efficiency.

**False Negative:** Pages that are actually declining may not be reviewed, leading to missed traffic and lost opportunities.

### Why Machine Learning?

Search performance depends on multiple interacting factors such as impressions, clicks, CTR, position, freshness, and engagement. Machine learning may help identify useful patterns that improve prioritization compared with simple fixed rules.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

df = pd.read_csv("/content/content_refresh_anonymized.csv")

print(df.head())

             content_id          client_id  search_volume  competition  \
0  content_304f48230142  client_f369cb89fc           10.0         0.67   
1  content_a1fb4e703a9e  client_4e07408562           90.0         0.01   
2  content_9aa793d4d895  client_7f2253d7e2            0.0         0.00   
3  content_331d6c4de07b  client_19581e27de           10.0         0.00   
4  content_d99b7a2d90ca  client_3fdba35f04            0.0         0.00   

  competition_level   cpc     content_type    main_intent  word_count  \
0              HIGH  2.05  keyword article  transactional      3221.0   
1               LOW  0.05  keyword article  informational      2481.0   
2               LOW  0.00  keyword article  informational      3515.0   
3               LOW  0.00  keyword article     commercial         NaN   
4               LOW  0.00  keyword article  informational      2803.0   

   char_count  ... char_count_tier   ctr  avg_position  engagement_rate  \
0     20457.0  ...     15000-25000  0.76 

In [ ]:
print("Rows:", len(df))
print("Columns:", len(df.columns))

print(df[["impressions_90d","clicks_90d","sessions_90d"]].describe())

Rows: 30000
Columns: 44
       impressions_90d    clicks_90d  sessions_90d
count     30000.000000  30000.000000  30000.000000
mean       5200.366300     16.097333     37.066633
std       16838.019547     75.076958    107.069131
min           1.000000      0.000000      1.000000
25%          81.000000      0.000000      2.000000
50%         731.000000      1.000000      7.000000
75%        3615.250000      7.000000     27.000000
max      517715.000000   4178.000000   4345.000000


In [ ]:
print(df["trend_direction"].value_counts())

print(df["content_type"].value_counts().head())

trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64


# 3. Quick Look at the Data

The starter dataset contains **30,000** content pages and **44** columns.

The average number of **90-day impressions is 5,200**, while the median is **731**, showing that page visibility varies substantially across the dataset.

The average number of **90-day sessions is 37**, with a median of **7**, indicating that many pages receive relatively low engagement while a smaller group attracts much higher traffic.

The target distribution also shows that **16,262 pages are labeled as "down"**, making decline detection and refresh prioritization an important decision-support problem for content teams.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# 4. Careful Words

This project identifies observed patterns in search and engagement data that may help prioritize pages for review.

The recommendations are intended to support human decision-making rather than replace it.

This analysis cannot prove that refreshing a page will increase search traffic or rankings.

It also cannot predict Google's ranking algorithm or establish causal relationships. Any recommendations should be validated by SEO specialists before implementation.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.